client/src/components/VideoCard.jsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+import Button from "./Button";
+
+const VideoCard = ({ videoData, setVideoData, singleVideo, onDelete }) => {
+  /**
+    On the return statement of the map, for the when the condition is true, I need to create the appropriate value that will replace the original value. When the condition is false I can return the video.
+   */
+
+  function upVoteHandler() {
+    const likeRating = videoData.map((video) => {
+      if (video.id === singleVideo.id) {
+        video.rating += 1;
+        fetch(`http://localhost:5000/videos/${video.id}`, {
+          method: "put",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify(video),
+        });
+      }
+      return video;
+    });
+    setVideoData(likeRating);
+  }
+
+  function downVoteHandler() {
```

Comment:
This function has a lot in common with the `upVoteHandler` - can you think how to make them share code, rather than having two copies of basically the same code?

Reply:
I have a new function in my level 300 called vote which I used to connect with both the upVote and downVote handlers.